model_zoo/odps_iris_dnn_model/odps_iris_dnn_model.py

```diff
@@ -17,13 +17,44 @@ def optimizer(lr=0.1):
     return tf.optimizers.SGD(lr)
 
 
-def dataset_fn(dataset, mode):
+def dataset_fn(dataset, mode, metadata):
     def _parse_data(record):
-        features = tf.reshape(record[0:3], (3, 1))
-        if mode == Mode.PREDICTION:
-            return features
+        label_col_name = "class"
+
+        def _get_features_without_labels(
+            record, label_col_ind, features_shape
+        ):
+            features = [
+                record[:label_col_ind],
+                record[label_col_ind + 1 :],  # noqa: E203
+            ]
+            features = tf.concat(features, -1)
+            return tf.reshape(features, features_shape)
+
+        features_shape = (4, 1)
+        labels_shape = (1,)
+        if mode != Mode.PREDICTION:
+            if label_col_name not in metadata.column_names:
+                raise ValueError(
+                    'Missing the label column "%s" in from the retrieved '
```

Comment:
Why not use `"` as the outer wrap, and `'%s'`?

Reply:
Good idea. Fixed so it's consistent with the string below